lib/migrator/methods/index.js

```diff
@@ -16,12 +16,21 @@ exports.rollback = require('./rollback');
 
 // Other API methods
 exports.getAllMigrationNames = require('./getAllMigrationNames');
+exports.getAllMigrationSourceNames = require('./getAllMigrationSourceNames');
```

Comment:
I think it would be better extend existing migrator method instead of adding new ones:

* getMigrationSourcePathByName(name) -> getMigrationPathByName(name, [migrationType])
* getMigrationSourceNameByPath(path) -> getMigrationNameByPath(path, [migrationType])
* getAllMigrationSourceNames() -> getAllMigrationNames([migrationType])
* isSourceDirExists() -> isDirExists([migrationType])
* isMigrationSourceExists(name) -> isMigrationExists([migrationType])

For all method `migrationType` is "executable" by default but could be set to "source". Internally such type could be resolved by some internal helper e.g. `_getMigrationTypeParams(migrationType)` which returns object with appropriate `{dir, extension}` or throws an error for unrecognized type.

In my opinion it also looks good for migration manager:

* manager.getMigrationSourcePath(name) -> getMigrationPath(name, [migrationType])

For me it looks like job is done with less effort and migrator, manager api are still good. What do you think?

Reply:
Yeah, think it is reasonable to make an enum `"executable" | "source"`.

Reply:
Done in https://github.com/okv/east/pull/54/commits/69649ef8b8694eb82d751a78cddbddea7761abc1